cleanlab/dataset.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Dataset Module
+Supports dataset-level and class-level automated quality, including finding which classes should
+be merged with other classes in your dataset, which classes should be removed,
+and which classes have the best label accuracy, and which have the worst labels.
+"""
+
+import numpy as np
+from count import estimate_joint, compute_confident_joint
+
+
+def find_similar_classes(
+    joint=None,
+    asymmetric=True,
+    *,
+    labels=None,
+    pred_probs=None,
+    confident_joint=None,
+    multi_label=False,
+):
+    """Ranks all pairs of classes by their similarity and returns the ordered pairs as well as
+    a similarity score between 0 and 1. Use this function to determine which classes in a dataset
+    should be merged.
+
+    The method uses the joint distribution of noisy and true labels to compute ontological issues
+    via the approach published in (Northcutt, Jiang, Chuang (2021) JAIR).
+    paper: https://arxiv.org/abs/1911.00068
+
+    This method works by providing any one (and only one) of the following inputs:
+    1. confident_joint
+    2. joint
+    3. labels and pred_probs
+    but it is not necessary to provide a combination, e.g. the labels and pred_probs and joint.
+
+    For parameter info, see the docstring of `count.estimate_joint`
```

Comment:
The docstring for `count.estimate_joint` does not seem to explain some of these args like `joint`, `asymmetric`. Furthermore the docstring of `count.estimate_joint` does not really explain its parameters either: 
https://docs.cleanlab.ai/master/cleanlab/count.html#cleanlab.count.estimate_joint
It just says:  See `cleanlab.count.calibrate_confident_joint` docstring.

Reply:
Even if `count.estimate_joint` did have good explanations of its parameters, I would prefer to explain the parameters separately here.  The reason is that `count.estimate_joint` seems like a more advanced function that requires understanding concepts, whereas `find_similar_classes` is supposed to be a super accessible function easily used by anybody.

Reply:
added params to both